DemoApp/DemoApp/DataAccessLayer/Services/PersonsDataService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using DataAccessLayer.Interfaces;
+using DataAccessLayer.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccessLayer.Services
+{
+    public class PersonsDataService : IPersonsDataService
+    {
+        private readonly SchoolContext _context;
+
+        public PersonsDataService(SchoolContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Person> CreatePersonAsync(Person newPerson) 
+        {
+            await _context.Persons.AddAsync(newPerson);
+            await _context.SaveChangesAsync();
+            return newPerson;
+        }
+
+        public async Task<Person?> GetPersonAsync(int personId)
+        {
+            return await _context.Persons.FirstOrDefaultAsync(person => person.Id == personId);
+        }
+
+        public async Task<IEnumerable<Person>> GetPersonsAsync()
```

Comment:
Why IEnumerable ?
You return a list right?

Reply:
changed to List